### DISCUSSION TITLE
Chrome-aws-lambda exceedes maximum size limit of 50mb

### DISCUSSION THREAD

**Discussion by Emir1999** [vercel/community#1136] [Help]
### Summary

I'm trying to deploy a code tha uses Puppeteer:
Here is my code: 
```javascript
import { addHttps } from '../../utils/onPageAnalyseUtils';

let chrome = {};
let puppeteer;

if (process.env.AWS_LAMBDA_FUNCTION_NAME) {
  puppeteer = require('puppeteer-core');
  chrome = require('chrome-aws-lambda');
} else {
  puppeteer = require('puppeteer');
}

const getTitle = async (req, res) => {
  let options = {};

  const url = req?.body?.url;
  const searchUrl = addHttps(url);

  if (process.env.AWS_LAMBDA_FUNCTION_VERSION) {
    options = {
      Rgs: [...chrome.args, '--hide-scrollbars', '--disable-web-security'],
      defaultViewport: chrome.defaultViewport,
      executablePath: await chrome.executablePath,
      headless: true,
      ignoreHTTPSErrors: true,
    };
  }

  try {
    const browser = await puppeteer.launch(options);

    const page = await browser.newPage();
    await page.goto(searchUrl);
    res.send(await page.title());
  } catch (error) {
    console.log(error);
    return null;
  }
};

export default getTitle;
```
Everything works fine in development but when I try to push it to vercel i get the following error:

![Unbenannt](https://user-images.githubusercontent.com/106652718/207282730-20ccb82e-ff59-48a7-a9ed-ca8dd581e4aa.PNG)

I've searched everywhere but not found an optimal solution.

**Comment by swarnava** [comment]
Indeed, the full puppeteer package cannot fit in a Serverless Function. The maximum size is [50MB](https://vercel.com/docs/v2/platform/limits#serverless-function-size)

You can consider using `puppeteer-core` instead.

There are also couple of workaround which you can find here: https://github.com/vercel/vercel/issues/4739#issuecomment-926879578

I will also recommend opening a thread for Next.js community: https://github.com/vercel/next.js/discussions
